README.md

```diff
@@ -1,2 +1,431 @@
+  >
+    <p>This action cannot be undone</p>
+    <Button text="Save" onChange={this.save}/>
+  </Modal>
+
+  // bad
+  <Modal
+    title="Confirm Changes"
+    open={open}
+    onClose={this.closeModal}>
+    <p>This action cannot be undone</p>
+    <Button text="Save" onChange={this.save}/>
+  </Modal>
+  ```
+* Self closing tags should not have closing bracket on its own line
+  ```js
+  // good
+  <Input
+    type="email"
+    label="Email"
+    placeholder="name@example.com"
+    onChange={this.setValue('email')}
+    onClick={this.selectText}/>
+
+  // bad
+  <Input
+    type="email"
+    label="Email"
+    placeholder="name@example.com"
+    onChange={this.setValue('email')}
+    onClick={this.selectText}
+  />
+  ```
+
+* Conditionally displayed elements should use the `&&` pattern. If the condition is not trivial it should be extracted.
+  ```js
+  <div>
+    {error && <div className="error">{error}</div>}
+  </div>
+  ```
+
+* Inline Map functions should use implicit return and parenthesis instead of explicit return and braces where possible.
+  ```js
+  // good
+  <ul>
+    {cars.map((car) => (
+      <li>{car.color}</li>
+    ))}
+  </ul>
+
+  // bad
+  <ul>
+    {cars.map((car) => {
+      return <li>{car.color}</li>;
+    })}
+  </ul>
+  ```
+
+## Whitespace
+
+* Object whitespace
+  ```js
+  // good
+  const car = {color: 'red', model: 'Saturn'};
+
+  // bad
+  const car = { color: 'red', model: 'Saturn' };
+  // bad
+  const car = {color:'red',model:'Saturn'};
+  ```
+
+* Import whitespace
+  ```js
+  // good
+  import { Card, Button } from 'semantic-ui-react';
+
+  // bad
+  import {Card, Button} from 'semantic-ui-react';
+  ```
+
+* Destructuring
+  ```js
+  // good
+  selectItem = ({target:{value:selectedItem=0}}) => this.setState({selectedItem})
+
+  // bad
+  selectItem = ({target: {value: selectedItem = 0}}) => this.setState({selectedItem})
+  // bad
+  selectItem = (e) => this.setState({selectedItem: e.target.value || 0})
+  ```
+
+* Collapse closing symbols
+  ```js
+  // good
+  cars.filter((car) => {
+    return car.color == 'red';
+  });
+
+  // less good
+  cars.filter(
+    (car) => {
+      return car.color == 'red';
+    }
+  );
+  ```
+
+## Methods
+
+* React component methods should be in [lifecycle order](https://reactjs.org/docs/react-component.html#the-component-lifecycle) with the exception of `render()` which should always be the last method.
+
+* Custom class methods should be between lifecycle and render.
+
+* Unless a constuctor is necessary, define initial state in the class scope
+  ```js
+  // good
+  class MyComponent extends React.Component {
+    state = {
+      active: true
+    }
+    ///...
+  }
+
+  // less good
+  class MyComponent extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        active: true
+      };
+    }
+    ///...
+  }
+  ```
+* If a method does not call another class method (such as `setState()`) extract it outside the class.
```

Comment:
what if the method does not setState but depends on a lot of props/state that would make the arity high